### PR TITLE
Add PostgreSQL support

### PR DESCRIPTION
**Pull request by Alan-Gomes**

### CODE REVIEW

**Comment by sgdc3** [comment]
https://codeclimate.com/github/AuthMe/AuthMeReloaded/pull/1620

**Comment by Alan-Gomes** [comment]
@sgdc3 
1- I've tested locally with the latest version of postgres (10.5) and it works like a charm

2- These warnings came from methods that I've copied from the MySQL class without any changes, that's weird

**Comment by sgdc3** [comment]
The parameters are simply ignored, please adapt them accordling to https://jdbc.postgresql.org/documentation/head/connect.html

**Comment by ljacqu** [comment]
I‘m not sure if it‘s possible with postgres but it would be awesome if we could have an integration test for the postgres variant as we do for the others.

**Comment by Alan-Gomes** [comment]
@sgdc3 I fixed the jdbc parameters according to the docs, some of them are'nt available on postgres :/

**Comment by Alan-Gomes** [comment]
@ljacqu I now created the tests according to the pattern of the project, I think it's right

**Comment by Alan-Gomes** [comment]
code climate is showing issues in methods that I'm not the author, I don't pretend to fix them to not create regression problems

**Comment by ljacqu** [comment]
To me everything looks great! Thanks for adding the tests. I think abstracting more of the common things between Postgres and MySQL isn't worth it since we already have quite a hierarchy in the data sources, plus we can't really do it in a natural way as soon as there is even a _minimal_ difference in syntax or parameters to set. 
Totally agree on not fixing issues you haven't introduced per se; in fact later on I wouldn't be against just suppressing those errors for the methods that are too long.

@sgdc3 The only thing I worry about is the addition of the postgres dependency to the pom? What's our strategy there?

**Comment by sgdc3** [comment]
@ljacqu sorry, what do you mean?

**Comment by ljacqu** [comment]
@sgdc3 Now postgres is a new dependency in the pom and I guess it will be shaded in, always. Is that an issue? I'm not on top of the POM stuff :sweat_smile:

**Comment by sgdc3** [comment]
We need to relocate it into our package, our jar is becoming too large, we should start working on a runtime dependency resolver.

**Comment by ljacqu** [comment]
OK, agree that we're starting to have an issue, shall we collect some idea and discuss sometime? I want to explore options before we start doing anything "weird" :D

**Comment by sgdc3** [comment]
@ljacqu sure! ;)

File: pom.xml [hunk 1]
```diff
@@ -808,6 +808,12 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
```
Comment (sgdc3) [inline pom.xml]: Please relocate the driver and its dependencies into our package in the maven shade plugin configuration.

Comment (Alan-Gomes) [reply pom.xml]: Done :)

**Comment by sgdc3** [comment]
@Alan-Gomes Thank you for your contribution! :)

**Pull request merged**